src/slack_helpers.py

```diff
@@ -418,3 +422,167 @@ def get_max_duration_block(cfg: config.Config) -> list[Option]:
+        # 99 is the limit for StaticSelectElement
+        # https://slack.dev/python-slack-sdk/api-docs/slack_sdk/models/blocks/block_elements.html#:~:text=StaticSelectElement(InputInteractiveElement)%3A%0A%20%20%20%20type%20%3D%20%22static_select%22-,options_max_length%20%3D%20100,-option_groups_max_length%20%3D%20100%0A%0A%20%20%20%20%40property%0A%20%20%20%20def%20attributes(
+        if len(groups) > 99:  # noqa: PLR2004
+            groups = groups[:99]
+        sorted_groups = sorted(groups, key=lambda groups: groups.name)
+        return InputBlock(
+            block_id=cls.GROUP_BLOCK_ID,
+            label=PlainTextObject(text="Select group"),
+            element=StaticSelectElement(
+                action_id=cls.GROUP_ACTION_ID,
+                placeholder=PlainTextObject(text="Select group"),
+                options=[Option(text=PlainTextObject(text=f"{group.name}"), value=group.id) for group in sorted_groups],
+            ),
+        )
+
+    @classmethod
+    def parse(cls, obj: dict) -> RequestForGroupAccess:  # noqa: ANN102
+        values = jp.search("view.state.values", obj)
+        hhmm = jp.search(f"{cls.DURATION_BLOCK_ID}.{cls.DURATION_ACTION_ID}.selected_option.value", values)
+        hours, minutes = map(int, hhmm.split(":"))
+        duration = timedelta(hours=hours, minutes=minutes)
+        return RequestForGroupAccess.parse_obj(
+            {
+                "permission_duration": duration,
+                "group_id": jp.search(f"{cls.GROUP_BLOCK_ID}.{cls.GROUP_ACTION_ID}.selected_option.value", values),
+                "reason": jp.search(f"{cls.REASON_BLOCK_ID}.{cls.REASON_ACTION_ID}.value", values),
+                "requester_slack_id": jp.search("user.id", obj),
+            }
+        )
+
+
+class ButtonGroupClickedPayload(BaseModel):
+    action: entities.ApproverAction
+    approver_slack_id: str
+    thread_ts: str
+    channel_id: str
+    message: dict
+    request: RequestForGroupAccess
+
+    class Config:
+        frozen = True
+
+    @root_validator(pre=True)
+    def validate_payload(cls, values: dict) -> dict:  # noqa: ANN101
+        fields = jp.search("message.blocks[?block_id == 'content'].fields[]", values)
+        requester_mention = cls.find_in_fields(fields, "Requester")
+        requester_slack_id = requester_mention.removeprefix("<@").removesuffix(">")
+        humanized_permission_duration = cls.find_in_fields(fields, "Permission duration")
+        permission_duration = unhumanize_timedelta(humanized_permission_duration)
+        group = cls.find_in_fields(fields, "Group")
+        group_id = group.split("#")[-1]
+        return {
+            "action": jp.search("actions[0].value", values),
+            "approver_slack_id": jp.search("user.id", values),
+            "thread_ts": jp.search("message.ts", values),
+            "channel_id": jp.search("channel.id", values),
+            "message": values.get("message"),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)

Suggested remediation:
Using the get method without a default argument to return the value of a key in a dictionary can cause undesirable results.
```
@@ -572,5 +572,5 @@
             &#34;thread_ts&#34;: jp.search(&#34;message.ts&#34;, values),
             &#34;channel_id&#34;: jp.search(&#34;channel.id&#34;, values),
-            &#34;message&#34;: values.get(&#34;message&#34;),
+            &#34;message&#34;: values.get(&#34;message&#34;, &#39;Enter default value&#39;),
             &#34;request&#34;: RequestForGroupAccess(
                 requester_slack_id=requester_slack_id,

```